File: WebApplication/src/modules/student/dtos/person-lot.dtos/person-student.dtos/student-semester.dto.ts
import { CourseType } from "src/modules/student/enums/course-type.enum";
import { Semester } from "src/modules/student/enums/semester.enum";
import { BaseSemesterDto } from "../base.dtos/base-semester.dto";
import { PeriodDto } from "../helpers/period.dto";

export class StudentSemesterDto extends BaseSemesterDto {
  period: PeriodDto;
  course: CourseType;
  semester: Semester;
}

File: StudentCard.Application/Students/Dtos/PersonLotDtos/PersonStudentDtos/StudentSemesterDto.cs
using StudentCard.Application.Students.Dtos.PersonLotDtos.BaseDtos;
using StudentCard.Data.Rdpzsd.Enums;

namespace StudentCard.Application.Students.Dtos.PersonLotDtos.PersonStudentDtos
{
    public class StudentSemesterDto : BaseSemesterDto
    {
        public PeriodDto Period { get; set; }
        public CourseType Course { get; set; }

        public Semester Semester { get; set; }
    }
}
